manager/catalog.go

```diff
@@ -86,9 +88,22 @@ func (cat *Catalog) readCatalog() error {
 }
 
 func (cat *Catalog) cloneCatalog() error {
-	log.Infof("Cloning the catalog from git URL %s", cat.URL)
+	var e *exec.Cmd
 	//git clone the repo
-	e := exec.Command("git", "clone", cat.URL, cat.catalogRoot)
+	// git clone -b mybranch --single-branch git://sub.domain.com/repo.git
+
+	if cat.URLBranch != "master" {
```

Comment:
pet peeve: `if ( !cond ) { a } else { b }` is confusing, flip the condition and the blocks (`if ( cond) { b } else { a }`)